db.go

```diff
@@ -86,9 +86,20 @@ type database struct {
 }
 
 type Options struct {
-	Path     string
+	// Path is the path to where the billy-files are stored. An empty value
+	// means 'memory-mode', in which nothing is stored to disk.
+	Path string
+	// MaxFileSize is the maximum size of files used. A value of zero means that
+	// there is no maximum.
+	MaxFileSize uint64
```

Comment:
Nitpick, you have MaxFileSize and ShelfFilesCount, but both refer to the same entity.

Perhaps rename to ShelfFileSize and ShelfFileCount (note, not plural in either case)